services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/IslCostSpec.groovy

```diff
@@ -158,4 +161,37 @@ class IslCostSpec extends BaseSpecification {
             assert islUtils.getIslInfo(links, isl.reversed).get().state == IslChangeType.DISCOVERED
         }
     }
+
+    @Tags(VIRTUAL)
+    def "ISL cost is NOT increased due to deactivating/activating switch"() {
+        given: "A switch"
+        def sw = topology.getActiveSwitches().first()
+
+        and: "Cost of related ISLs"
+        def swIsls = topology.getRelatedIsls(sw)
+        def swIslsCostMap = swIsls.collectEntries { isl ->
+            [isl, islUtils.getIslInfo(isl).get().cost]
+        }
+
+        when: "Deactivate the switch"
+        lockKeeper.knockoutSwitch(sw)
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            assert northbound.getSwitch(sw.dpId).state == SwitchChangeType.DEACTIVATED
+            def links = northbound.getAllLinks()
+            swIsls.each { assert islUtils.getIslInfo(links, it).get().state == IslChangeType.FAILED }
+        }
```

Comment:
It would be nice to verify that ISL cost was not increased at this point, before activating the switch back